src/monodroid/jni/debug.cc

```diff
@@ -213,14 +211,12 @@ Debug::start_debugging_and_profiling ()
 	char *connect_args;
 	androidSystem.monodroid_get_system_property (Debug::DEBUG_MONO_CONNECT_PROPERTY, &connect_args);
 
-	if (connect_args != nullptr) {
-		int res = start_connection (connect_args);
-		if (res != 2) {
-			if (res) {
-				log_fatal (LOG_DEBUGGER, "Could not start a connection to the debugger with connection args '%s'.", connect_args);
-				exit (FATAL_EXIT_DEBUGGER_CONNECT);
-			}
-
+	if (connect_args != nullptr && strcmp (connect_args, "") != 0) {
```

Comment:
This introduces a memory leak, as we'll skip the `delete[] connect_args` (line 227) if `connect_args` is the empty string.